app/src/main/java/org/mozilla/fenix/utils/ToolbarPopupWindow.kt

```diff
@@ -35,12 +38,13 @@ object ToolbarPopupWindow {
         copyVisible: Boolean = true,
     ) {
         val context = view.get()?.context ?: return
+        val isCustomTabSession = customTabId != null
         val clipboard = context.components.clipboardHandler
-        val clipboardUrl = clipboard.getUrl()
-        val clipboardText = clipboard.text
-        if (!copyVisible && clipboardUrl == null) return
 
-        val isCustomTabSession = customTabId != null
+        val containsText = clipboard.containsText()
+        val containsUrl = clipboard.containsURL()
+
+        if (!copyVisible && (isCustomTabSession || (!containsText && !containsUrl))) return
```

Comment:
We got a lint error for complex condition here. Logs available at https://github.com/mozilla-mobile/fenix/actions/runs/3272855315/jobs/5827687650#step:4:169. We will probably want to group some of these conditionals into a variable.